### PR TITLE
issue: 26871 - Remove SimplePath class from patches.py

### PR DESCRIPTION
**Pull request by prateetishah**
## PR summary
This PR removes the deprecated code during the version 3.7 from lib/matplotlib/patches.py which is a sub issue in the parent issues here: https://github.com/matplotlib/matplotlib/issues/26865
## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: doc/api/next_api_changes/removals/00001-ABC.rst [hunk 1]
```diff
@@ -1,7 +0,0 @@
-Removal change template
```
Comment (tacaswell) [inline doc/api/next_api_changes/removals/00001-ABC.rst:1]: Please don't delete the template!

File: doc/api/next_api_changes/removals/26871-PS.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Remove Deprecated API
+~~~~~~~~~~~~~~~~~~~~~~~
+``matplotlib.patches.SimplePath``
```
Comment (tacaswell) [inline doc/api/next_api_changes/removals/26871-PS.rst:3]: Can you check the deprecation note an copy over the alternative if any?

**Comment by tacaswell** [comment]
Can you please squash these commits?

https://matplotlib.org/devdocs/devel/development_workflow.html#rewrite-commit-history

**Review by Vallabh100 — APPROVED** [review]

**Comment by ksunden** [comment]
The git history here seems to be messy and includes unrelated commits from e.g. #26505, but overwrites their commit messages. 

We will also need to sort out between this and #26890, which also is doing the same deprecation task (and while that author closed/reopened several PRs, did get the first one as #26875, if we use that as the metric)

**Comment by ksunden** [comment]
Thank you for your efforts here, however we elected to merge another PR that took on this portion of this task. We hope to hear from you again

**Pull request closed without merging**